api/albums-controller.js

```diff
@@ -59,14 +62,24 @@ async function joinAlbum(userToken, userId, shareToken) {
   return result;
 }
 
+async function joinPendingAlbums(userToken, userId) {
+  const shareTokens = getShareTokens(userId);
+  const joinAllPromises = shareTokens.map(token => {
+    return async () => {
+      return joinAlbum(userToken, userId, token);
```

Comment:
If joinAlbum already returns a promise, no need to wrap this function call in another function.